cmake/settings.cmake

```diff
@@ -0,0 +1,55 @@
+# Copyright (c) Microsoft Corporation
+# SPDX-License-Identifier: MIT
+
+# Some targets do not play well with the default definitions (such
+# as bpftool and /DWIN32). Remove them from the variables for now
+set(settings_variable_list
+  "CMAKE_C_FLAGS_RELEASE"
+  "CMAKE_C_FLAGS_DEBUG"
+  "CMAKE_C_FLAGS_RELWITHDEBINFO"
+
+  "CMAKE_CXX_FLAGS_RELEASE"
+  "CMAKE_CXX_FLAGS_DEBUG"
+  "CMAKE_CXX_FLAGS_RELWITHDEBINFO"
+
+  "CMAKE_C_FLAGS"
+  "CMAKE_CXX_FLAGS"
+)
+
+foreach(settings_variable ${settings_variable_list})
+  string(REPLACE "/D_WINDOWS" "" "${settings_variable}" ${${settings_variable}})
+  string(REPLACE "/DWIN32" "" "${settings_variable}" ${${settings_variable}})
+endforeach()
+
+add_library("ebpf_for_windows_base_definitions" INTERFACE)
+target_compile_definitions("ebpf_for_windows_base_definitions" INTERFACE
+  $<$<CONFIG:Debug>:_DEBUG>
+  $<$<CONFIG:Release>:NDEBUG>
+  $<$<CONFIG:RelWithDebInfo>:NDEBUG>
```

Comment:
FYI the ebpf-verifier project doesn't seem to need this and it gets NDEBUG in the generated PreprocessorDefinitions just fine.  Just wondering why it's needed here (not a blocking comment).

Reply:
They have been made explicit here in case we want to use a different generator in the future (such as Ninja build, which has a huge performance boost in parallel builds). The target itself is something I usually implement in order to easily change common settings on all the project targets